documentation/12-factor-app-devon4j.asciidoc

```diff
@@ -0,0 +1,151 @@
+:toc: macro
+toc::[]
+:idprefix:
+:idseparator: -
+
+= 12-factor-app with devon4j
+
+This document mainly focuses on discussing how can you create 12 factor app with devon4j. To know more about this 12 factors you can refer https://12factor.net/[here] . Twelve factor is mainly focus on creating cloud native applications. These are the guidelines on what factors you need to consider in different stages of application lifecycle.
+
+== Codebase
+
+*One codebase tracked in revision control, many deploys*
+
+First factor to consider while creating cloud native application is 1 codebase = 1 application = 1 process. 
+
+Traditionally we are used to develop and maintain monolithic applications. But it is difficult to deploy monolith appication as it is into the cloud environment it might night give us a benefit of cloud if we just do lift and shift. So we need to change our application architecture to adapt it like a cloud applications. One of the way to consider is breaking your application into multiple microservices. For simple application it might not be applicable but for some complex system user can check if they want to go ahead with microservices approach.
+To satisfy this factor consider creating one git/code repository for 1 process.
+
+== Dependencies
+
+*Explicitly declare and isolate dependencies*
+
+A twelve-factor app never relies on implicit existence of system-wide packages. It should explicitly declare and isolate dependencies. This is required to make repeatable deployments of an app without building it again and again.
+
+devon4j application is a maven project so we declare dependencies in pom.xml. For isolation of dependencies we should containerize an application.It will isolate application environment from rest of the environment.
```

Comment:
Here you are missing a space between the last sentences.

Also, we should add a link on how to put the application in a container. For both quarkus and spring